src/mysql-util/src/aws_rds.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use aws_config::{Region, SdkConfig};
+use aws_credential_types::provider::error::CredentialsError;
+use aws_credential_types::provider::ProvideCredentials;
+use aws_sigv4::http_request::{
+    self, SignableBody, SignableRequest, SignatureLocation, SigningError, SigningSettings,
+};
+use aws_sigv4::sign::v4::signing_params::BuildError;
+use aws_sigv4::sign::v4::SigningParams;
+use std::time::Duration;
+
+const AWS_TOKEN_EXPIRATION_SECONDS: u64 = 900;
+const AWS_SERVICE: &str = "rds-db";
+const DB_ACTION: &str = "connect";
+
+#[derive(Debug, thiserror::Error)]
+pub enum RdsTokenError {
+    /// AWS SdkConfig did not contain a credentials provider
+    #[error("Credentials provider required to sign RDS IAM request")]
+    MissingCredentialsProvider,
+
+    /// Credentials provider failed to generate credentials to use for signing
+    #[error(transparent)]
+    CredentialsError(#[from] CredentialsError),
+
+    /// Invalid options for signing the token request
+    #[error(transparent)]
+    SigningParametersBuildError(#[from] BuildError),
+
+    /// Failed to parse the url or create a signed request
+    #[error(transparent)]
+    SigningError(#[from] SigningError),
+}
+
+// Generate an RDS authentication token.  This should mirror what can be found
+// in aws_sdk_rds::auth_token, but without restricted creates.
+pub(crate) async fn rds_auth_token(
+    host: &str,
+    port: u16,
+    username: &str,
+    aws_config: &SdkConfig,
+) -> Result<String, RdsTokenError> {
+    let credentials = aws_config
+        .credentials_provider()
+        .ok_or(RdsTokenError::MissingCredentialsProvider)?
+        .provide_credentials()
+        .await?;
+
+    // mirror existing SDK behaviour: default to us-east-1,
+    let region = aws_config
+        .region()
+        .cloned()
+        .unwrap_or_else(|| Region::new("us-east-1"));
+
+    // defaults to SystemTime
+    let time_source = aws_config.time_source().unwrap_or_default();
+
+    let mut signing_settings = SigningSettings::default();
+    signing_settings.expires_in = Some(Duration::from_secs(AWS_TOKEN_EXPIRATION_SECONDS));
```

Comment:
Not blocking, but feels like it might be nice to make this a dyncfg, so it can live configure it with LaunchDarkly if we ever hit timeouts. Here are some [examples](https://github.com/MaterializeInc/materialize/blob/main/src/adapter-types/src/dyncfgs.rs)

Reply:
No objection at all, merge away!